### DISCUSSION TITLE
Invalidate atomWithQuery

### DISCUSSION THREAD

**Discussion by mwisnicki** [pmndrs/jotai#1641] [Ideas]
Currently [AtomWithQueryAction ](https://github.com/pmndrs/jotai/blob/main/src/query/atomWithQuery.ts#L8) only allows immediate refetching via:

```
set(anAtom, {type: "refetch"});
```

It would be useful to also support "invalidate" as I feel that's more appropriate action most of the time.

**Comment by dai-shi** [comment]
what would be the behavioral difference?

**Reply by mwisnicki** [reply]
My understanding was that refetch is always executed while invalidate just marks resource as stale and only refetches when it is being used.
I have a dialog/popover that has an action which should invalidate the query and close it but the query does not need to be refetched until dialog is reopened.

**Reply by dai-shi** [reply]
Yeah, that makes sense. Apart from the query integration, from the atom model perspective, if an atom is not subscribed, it will not refetch immediately. So, behavior-wise, it's "invalidate".

atomsWithQuery from jotai-tanstack-query has two ways of "refetch" (force: true/false). Ironically, force=true behavior would be close to "invalidate".

On second thought, we would need to try, but if we did "invalidate" a query actually in TanStack Query, we might have no way to actually refetch...
Can you play a bit and see how it would work?
